packages/path_provider/path_provider_windows/example/lib/main.dart

```diff
@@ -20,10 +20,10 @@ class MyApp extends StatefulWidget {
 }
 
 class _MyAppState extends State<MyApp> {
-  String _tempDirectory = 'Unknown';
-  String _downloadsDirectory = 'Unknown';
-  String _appSupportDirectory = 'Unknown';
-  String _documentsDirectory = 'Unknown';
+  String? _tempDirectory = 'Unknown';
+  String? _downloadsDirectory = 'Unknown';
+  String? _appSupportDirectory = 'Unknown';
+  String? _documentsDirectory = 'Unknown';
```

Comment:
I wonder about the choice of path provider returning `null` for things like `provider.getTemporaryPath()`. I can't think of a standard library that returns `null` for the temporary directory.  

Should those cases throw instead? Async methods are meant to be enclosed in a try/catch at some point.

Reply:
Well apparently it made sense to the reviewer of https://github.com/flutter/plugins/pull/3249 😉 

I think it's a bit tricky; some of these can definitely return null on some platforms, others *probably* can't, but I don't know that every platform actually strictly guarantees a valid return value. I can definitely see the argument that if someone is using path_provider to get the temp path, or the app support directory, and it's null, they likely don't have any useful fallback they could do, so making them handle null is likely not a great experience. I'm not 100% sure where to draw that line though. Is it just those two?

I think if we do want to make some of these non-nullable but throwing, we should do it in the app-facing package, and leave the platform interface as-is, so that we're ensuring consistency in that regardless of platform and implementation. Let's discuss in chat, and when we do the stable release of `path_provider` we can make that change there if we want to.